federation/pkg/kubefed/init/init_test.go

```diff
@@ -106,26 +106,28 @@ func TestInitFederation(t *testing.T) {
 		dryRun                       string
 		apiserverArgOverrides        string
 		cmArgOverrides               string
+		controllerOptionalSecrets    map[string]string
 		apiserverEnableHTTPBasicAuth bool
 		apiserverEnableTokenAuth     bool
 		isRBACAPIAvailable           bool
 	}{
 		{
-			federation:            "union",
-			kubeconfigGlobal:      fakeKubeFiles[0],
-			kubeconfigExplicit:    "",
-			dnsZoneName:           "example.test.",
-			lbIP:                  lbIP,
-			apiserverServiceType:  v1.ServiceTypeLoadBalancer,
-			serverImage:           "example.test/foo:bar",
-			etcdPVCapacity:        "5Gi",
-			etcdPersistence:       "true",
-			expectedErr:           "",
-			dnsProvider:           util.FedDNSProviderCoreDNS,
-			dnsProviderConfig:     "dns-provider.conf",
-			dryRun:                "",
-			apiserverArgOverrides: "--client-ca-file=override,--log-dir=override",
-			cmArgOverrides:        "--dns-provider=override,--log-dir=override",
+			federation:                "union",
+			kubeconfigGlobal:          fakeKubeFiles[0],
+			kubeconfigExplicit:        "",
+			dnsZoneName:               "example.test.",
+			lbIP:                      lbIP,
+			apiserverServiceType:      v1.ServiceTypeLoadBalancer,
+			serverImage:               "example.test/foo:bar",
+			etcdPVCapacity:            "5Gi",
+			etcdPersistence:           "true",
+			expectedErr:               "",
+			dnsProvider:               util.FedDNSProviderCoreDNS,
+			dnsProviderConfig:         "dns-provider.conf",
+			dryRun:                    "",
+			apiserverArgOverrides:     "--client-ca-file=override,--log-dir=override",
+			cmArgOverrides:            "--dns-provider=override,--log-dir=override",
+			controllerOptionalSecrets: map[string]string{"key1": "val1"},
```

Comment:
better to use strings to indicate the actual values, like `mountpath` instead of `val1`.